.github/workflows/build.yml

```diff
@@ -6,7 +6,6 @@ name: Build
 # Run this Build only for pushes / PRs to main branch
 on:
   push:
-    branches: main
```

Comment:
@benbosman : Good catch here, however I don't think we should merge this PR *as is*.  The reason is that you've enabled this build *on every branch*, which would include our maintenance branches (`dspace-6_x`, `dspace-5_x`, etc).  That would mean that new commits to those branches would end up running the DSpace 7 build process...which is not what we want.

Instead, I think we want this setting to be:
```
on:
  push:
    # Ignore any maintenance branches, e.g. dspace-6_x and older
    branches-ignore: **/dspace-**
```

See does on ignoring branches at https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-ignoring-branches-and-tags

Reply:
I've realized the above is slightly flawed...as technically, we don't know if a branch named `w2p-74727_Nullpointer-workflow-item-after-submitter-delete` should run the `main` build, the `dspace-6_x` build or the `dspace-5_x` build.  In the above example, I'm assuming **ALL** local branches run the `main` build, but that's not likely to be the case.  So, my suggested solution may not be ideal either.

Reply:
@abollini pointed out via Slack that my assumptions above seem to be incorrect.  GitHub Actions is only using the `build.yml` that exists _in your current branch_.  (I'm not sure how I got in my head that it didn't work this way, but I'm glad to be wrong, as this makes our configuration a ton easier to figure out)

So, this means, the ideal configuration for us would NOT limit by branch name, and would just be:
```
on: [push, pull_request]
```